eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/OfferTypes.scala

```diff
@@ -209,37 +190,36 @@ object OfferTypes {
   }
 
   case class InvoiceRequest(records: TlvStream[InvoiceRequestTlv]) {
-    val chain: ByteVector32 = records.get[Chain].map(_.hash).getOrElse(Block.LivenetGenesisBlock.hash)
-    val offerId: ByteVector32 = records.get[OfferId].map(_.offerId).get
-    val amount: Option[MilliSatoshi] = records.get[Amount].map(_.amount)
-    val features: Features[InvoiceFeature] = records.get[FeaturesTlv].map(_.features.invoiceFeatures()).getOrElse(Features.empty)
-    val quantity_opt: Option[Long] = records.get[Quantity].map(_.quantity)
+    val offer: Offer = Offer.validate(TlvStream[OfferTlv](records.records.collect { case tlv: OfferTlv => tlv }, records.unknown.filter(_.tag < UInt64(80)))).toOption.get
+
+    val metadata: Option[ByteVector] = records.get[InvoiceRequestMetadata].map(_.data)
+    val chain: ByteVector32 = records.get[InvoiceRequestChain].map(_.hash).getOrElse(Block.LivenetGenesisBlock.hash)
+    val amount: Option[MilliSatoshi] = records.get[InvoiceRequestAmount].map(_.amount)
+    val features: Features[InvoiceFeature] = records.get[InvoiceRequestFeatures].map(_.features.invoiceFeatures()).getOrElse(Features.empty)
+    val quantity_opt: Option[Long] = records.get[InvoiceRequestQuantity].map(_.quantity)
     val quantity: Long = quantity_opt.getOrElse(1)
-    val payerKey: ByteVector32 = records.get[PayerKey].get.publicKey
-    val payerNote: Option[String] = records.get[PayerNote].map(_.note)
-    val payerInfo: Option[ByteVector] = records.get[PayerInfo].map(_.info)
-    val replaceInvoice: Option[ByteVector32] = records.get[ReplaceInvoice].map(_.paymentHash)
-    val payerSignature: ByteVector64 = records.get[Signature].get.signature
+    val payerId: PublicKey = records.get[InvoiceRequestPayerId].get.publicKey
+    val payerNote: Option[String] = records.get[InvoiceRequestPayerNote].map(_.note)
+    private val signature: ByteVector64 = records.get[Signature].get.signature
 
-    def isValidFor(offer: Offer): Boolean = {
+    def isValidFor(otherOffer: Offer): Boolean = {
```

Comment:
Same as for the similar function in `Bolt12Invoice`, when we receive an invoice request, we need to check that it is valid for our offer.

Reply:
But it is backwards, isn't it? When we receive an invoice request, we will look at the enclosed offer and check the local DB to see if we have a matching offer that hasn't expired (we don't have "our offer" somehow pre-loaded)? So that check will be in the method that scans the DB, not a separate method we call afterwards?

Reply:
Not necessarily, if we use blinded paths, we will receive an invoice request to a specific path_id and retrieve the offer matching the path_id.